contracts/proxy/IdentityProxy.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.6.9;
+
+
+interface IImplementationAuthority {
```

Comment:
why do you declare it here? i would do an external contract and i would put the updateImplementation function on the interface as well

Reply:
This is to keep the IdentityProxy as a standalone and flat contract. The IdentityProxy only needs the getImplementation method.